README.md

```diff
@@ -56,186 +50,318 @@ npm install @dojo/has
 
-```tsx
-class MyWidgetWithTsx extends WidgetBase<MyProperties> {
-	protected render(): DNode {
-		const { clear, properties: { completed, count, activeCount, activeFilter } } = this;
+### Classes & Theming
 
-		return (
-			<footer classes={this.classes(css.footer)}>
-				<span classes={this.classes(css.count)}>
-					<strong>{`${activeCount}`}</strong>
-					<span>{`${count}`}</span>
-				</span>
-				<TodoFilter activeFilter={activeFilter} />
-				{ completed ? ( <button onclick={clear} /> ) : ( null ) }
-			</footer>
-		);
-	}
-}
-```
+Section coming soon!
 
-**Note:** Unfortunately `tsx` is not directly used within the module so will report as an unused import so would be needed to be ignored by linters.
+### Internationalization
 
-### Writing Custom Widgets
+Widgets can be internationalized by adding the `I18nMixin` mixin from `@dojo/widget-core/mixins/I18n`. [Message bundles](https://github.com/dojo/i18n) are localized by passing them to `localizeBundle`.
 
-The `WidgetBase` class provides the functionality needed to create Custom Widgets.
-This functionality includes caching and widget lifecycle management.
+If the bundle supports the widget's current locale, but those locale-specific messages have not yet been loaded, then the default messages are returned.
+The widget will be invalidated once the locale-specific messages have been loaded.
 
-The `WidgetBase` class is available from the `@dojo/widget-core/WidgetBase ` package.
+Each widget can have its own locale by passing a property - `properties.locale`.
+If no locale is set, then the default locale, as set by [`@dojo/i18n`](https://github.com/dojo/i18n), is assumed.
 
 ```ts
-import { WidgetBase } from '@dojo/widget-core/WidgetBase';
+const MyWidgetBase = I18nMixin(WidgetBase);
+
+class I18nWidget extends MyWidgetBase<I18nWidgetProperties> {
+	render: function () {
+		// Load the "greetings" messages for the current locale. If the locale-specific
+		// messages have not been loaded yet, then the default messages are returned,
+		// and the widget will be invalidated once the locale-specific messages have
+		// loaded.
+		const messages = this.localizeBundle(greetingsBundle);
+
+		return v('div', { title: messages.hello }, [
+			w(Label, {
+				// Passing a message string to a child widget.
+				label: messages.purchaseItems
+			}),
+			w(Button, {
+				// Passing a formatted message string to a child widget.
+				label: messages.format('itemCount', { count: 2 })
+			})
+		]);
+	}
+}
 ```
 
-**All** widgets should extend from this class.
+## Key Principles
+
+These are some of the **important** principles to keep in mind when creating and using widgets:
 
-#### The 'properties' lifecycle
+1. The widget's *`__render__`*, *`__setProperties__`*, *`__setChildren__`* functions should **never** be called or overridden
```

Comment:
some of these are sentences with a full stop, some aren't... should be consistent

Also, if you expounded on _why_ some of these are principles... like:

1. The widget's *`__render__`*, *`__setProperties__`*, *`__setChildren__`* functions should **never** be called or overridden.  These are the internal methods of the widget APIs and their behaviour can change in the future, causing regressions in your application.

It is human nature to do something you are told not to do without justification. 😈